src/model/game/GameManager.ts

```diff
@@ -836,23 +26,54 @@ export default class GameManager {
 
     // Used for loading game specific settings before game is selected.
     static get defaultGame(): Game {
-        return this._gameList.find(value => value.internalFolderName === "RiskOfRain2")!;
+        return this.gameList.find(value => value.internalFolderName === "RiskOfRain2")!;
     }
 
     static get gameList(): Game[] {
-        return [...this._gameList];
+        return EcosystemSchema.r2mm_games.map((game) => {
+            const r2mm = game.r2modman!;
+
+            // Map distribution values onto the r2mm StorePlatformMetadata type.
+            // The `identifier` property may not be present, so we check it beforehand to avoid JSON import lint errors.
+            const distributions = game.distributions.map((x: any) => new StorePlatformMetadata(
+                    GetStorePlatformFromName(x.platform),
+                    x.identifier!
+                )
+            );
+
+            const displayMode = (GameSelectionDisplayMode as any)[(r2mm.gameSelectionDisplayMode as string).toUpperCase()];
```

Comment:
AFAICT this doesn't work and eradicates type safety. I'd create a helper, maybe in GameSelectionDisplayMod.ts:

```
export function displayModeFromString(mode: string): GameSelectionDisplayMode {
    if (mode.toLowerCase() === "visible") return GameSelectionDisplayMode.VISIBLE;
    if (mode.toLowerCase() === "hidden") return GameSelectionDisplayMode.HIDDEN;
    throw new Error(`Invalid GameSelectionDisplayMode "${mode}"`);
}
```

Something similar should be done for GameInstanceType.